benchmark/test_benchmark_jubjub.py

```diff
@@ -0,0 +1,151 @@
+        test_preprocessing.generate(kind, n, t, k=k)
+
+    program_runner = TaskProgramRunner(n, t, CONFIG)
+    program_runner.add(_prog)
+    loop = asyncio.get_event_loop()
+
+    def _work():
+        loop.run_until_complete(program_runner.join())
+
+    benchmark(_work)
+
+
+def test_benchmark_shared_point_double(test_preprocessing, benchmark):
+    async def _prog(context):
+        result = SharedPoint.from_point(context, TEST_POINT)
+        result = result.double()
+        await result.open()
+
+    for kind in TEST_PREPROCESSING:
+        test_preprocessing.generate(kind, n, t, k=k)
+
+    program_runner = TaskProgramRunner(n, t, CONFIG)
+    program_runner.add(_prog)
+    loop = asyncio.get_event_loop()
+
+    def _work():
+        loop.run_until_complete(program_runner.join())
+
+    benchmark(_work)
+
+
+# Following tests are commented out for now, as they hang currently.
+
+@mark.parametrize("multiplier", [1, 4, 1024 ** 3, TEST_CURVE.Field.modulus])
+def test_benchmark_shared_point_mul(test_preprocessing, benchmark, multiplier):
+    async def _prog(context):
+        base = SharedPoint.from_point(context, TEST_POINT)
+        result = base.mul(multiplier)
+        await result.open()
+
+    for kind in TEST_PREPROCESSING:
+        test_preprocessing.generate(kind, n, t, k=k)
+
+    program_runner = TaskProgramRunner(n, t, CONFIG)
+    program_runner.add(_prog)
+    loop = program_runner.loop
+
+    def _work():
+        loop.run_until_complete(program_runner.join())
+
+    benchmark(_work)
+
+
+@mark.parametrize("multiplier", [1, 4, 1024 ** 3, TEST_CURVE.Field.modulus])
+def test_benchmark_shared_point_montgomery_mul(
+        test_preprocessing, benchmark, multiplier):
+    async def _prog(context):
+        base = SharedPoint.from_point(context, TEST_POINT)
+        result = base.montgomery_mul(multiplier)
+        await result.open()
+
+    for kind in TEST_PREPROCESSING:
+        test_preprocessing.generate(kind, n, t, k=k)
+
+    program_runner = TaskProgramRunner(n, t, CONFIG)
+    program_runner.add(_prog)
+    loop = asyncio.get_event_loop()
+
+    def _work():
+        loop.run_until_complete(program_runner.join())
+
+    benchmark(_work)
+
+
+@mark.parametrize("bit_length", [0, 40, 80, 160, 256])
+def test_benchmark_share_mul(bit_length, test_preprocessing, benchmark):
+    async def _prog(context):
+        # First round is just for preprocessing
+        if bit_length == 0:
+            return
+
+        p = TEST_POINT
+
+        multiplier_ = Jubjub.Field(0)
+        m_bits = [test_preprocessing.elements.get_bit(context)
+                  for i in range(bit_length)]
+
+        for idx, m in enumerate(m_bits):
+            multiplier_ += (2**idx) * m
+
+        # Compute share_mul
+        p1_ = await share_mul(context, m_bits, p)
+        result = await p1_.open()
+
+        # Assertation
+        multiplier = await multiplier_.open()
+        if multiplier == Jubjub.Field(0):
+            assert result == p
```

Comment:
You should run the benchmark code (to get the best results) with the `-O` flag which disables the asserts so I would advise not having asserts here. If you want to test something then you should add it as a test and not as a benchmark.

Reply:
Good point-- stripped them out.